### PR TITLE
Moderation > Make Block and Report button less prominent in the frontend

### LINKED ISSUE #2222: Moderation > Make Block and Report button less prominent in the frontend
**Task 1**
Moderation Report/Block button should be less prominent everywhere.
Make it look more subtle.
We can show all buttons in the ellipsis dropdown.

**Check video below for more details**
https://www.loom.com/share/ac4240165fa14f8491b3259519b451c6

---

**Related Task 2**
Rename the "Report" button everywhere in the dropdown and call it "Report Content_Type"

**Again, check video below for more details**
https://www.loom.com/share/bf1e3c76c5634c7cac53b83be8c8af68

---

**Support Links:**
https://www.facebook.com/groups/buddyboss/permalink/1171540939947413/

### PR DESCRIPTION
**Pull request by amincdev**
### All Submissions:

* [ ] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [ ] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Connected PR in Theme
https://github.com/buddyboss/buddyboss-theme/pull/1343

### Changes proposed in this Pull Request:

Fixes #2222 .

### Proof Screenshots or Video

https://loom.com/share/64c1ec54d5624cb3b8f61ef1db41698f

### Other information ( Need to test below points ):
- Blog comment should have more options dropdown just like other places
- Add missing Forum Report button in platform
- I don’t think we should change the text to “Reported” once a user has made a report. I know this is the current way the buttons work, but it’s bad UX to have a button/link that does nothing,
	- It should always say “Report [Type]”
	- When the click on it again, it should open a modal that says: Report [Type]
 	- You have already reported this **[type].**

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Comment by janakkaneriya007** [comment]
@amincdev I have finished backend changes you can design it new changes

File: src/bp-moderation/bp-moderation-functions.php [hunk 1]
```diff
@@ -1086,3 +1071,81 @@ function bp_moderation_item_count( $args = array() ) {
...
+
+	/**
+	 * Filters the reported content type
+	 *
+	 * @since BuddyBoss X.X.X
+	 *
+	 * @param string $content_type Content type.
+	 * @param int    $item_id      Item id.
+	 */
+	return apply_filters( "bp_moderation_{$item_type}_report_content_type", __( 'Post', 'buddyboss' ), $item_id );
```
Comment (KartikSuthar) [inline src/bp-moderation/bp-moderation-functions.php:1098]:
```suggestion
	return apply_filters( "bp_moderation_{$item_type}_report_content_type", esc_html__( 'Post', 'buddyboss' ), $item_id );
```

File: src/bp-moderation/bp-moderation-functions.php [hunk 2]
```diff
@@ -1086,3 +1071,81 @@ function bp_moderation_item_count( $args = array() ) {
...
+
+	/**
+	 * Filters the report button text for different components
+	 *
+	 * @since BuddyBoss 1.7.2
+	 *
+	 * @param string $button_text Button text.
+	 * @param int    $item_id     Item id.
+	 */
+	return apply_filters( "bb_moderation_{$item_type}_report_button_text", __( 'Report', 'buddyboss' ), $item_id );
```
Comment (KartikSuthar) [inline src/bp-moderation/bp-moderation-functions.php:1124]:
```suggestion
	return apply_filters( "bb_moderation_{$item_type}_report_button_text", esc_html__( 'Report', 'buddyboss' ), $item_id );
```

File: src/bp-moderation/bp-moderation-functions.php [hunk 3]
```diff
@@ -1086,3 +1071,81 @@ function bp_moderation_item_count( $args = array() ) {
...
+
+	/**
+	 * Filters the reported button text for different components
+	 *
+	 * @since BuddyBoss 1.7.2
+	 *
+	 * @param string $button_text Button text.
+	 * @param int    $item_id     Item id.
+	 */
+	return apply_filters( "bb_moderation_{$item_type}_reported_button_text", __( 'Reported', 'buddyboss' ), $item_id );
```
Comment (KartikSuthar) [inline src/bp-moderation/bp-moderation-functions.php:1150]:
```suggestion
	return apply_filters( "bb_moderation_{$item_type}_reported_button_text", esc_html__( 'Reported', 'buddyboss' ), $item_id );
```

File: src/bp-moderation/classes/class-bp-moderation-activity-comment.php [hunk 4]
```diff
@@ -226,4 +234,32 @@ protected function is_content_hidden( $item_id ) {
...
+	 *
+	 * @since BuddyBoss X.X.X
+	 *
+	 * @param string $button_text Button text.
+	 * @param int    $item_id     Item id.
+	 *
+	 * @return string
+	 */
+	public function report_button_text( $button_text, $item_id ) {
+		return __( 'Report Comment', 'buddyboss' );
```
Comment (KartikSuthar) [inline src/bp-moderation/classes/class-bp-moderation-activity-comment.php:249]:
```suggestion
		return esc_html__( 'Report Comment', 'buddyboss' );
```

File: src/bp-moderation/classes/class-bp-moderation-activity-comment.php [hunk 5]
```diff
@@ -226,4 +234,32 @@ protected function is_content_hidden( $item_id ) {
...
+	 *
+	 * @since BuddyBoss X.X.X
+	 *
+	 * @param string $content_type Button text.
+	 * @param int    $item_id     Item id.
+	 *
+	 * @return string
+	 */
+	public function report_content_type( $content_type, $item_id ) {
+		return __( 'Comment', 'buddyboss' );
```
Comment (KartikSuthar) [inline src/bp-moderation/classes/class-bp-moderation-activity-comment.php:263]:
```suggestion
		return esc_html__( 'Comment', 'buddyboss' );
```

File: src/bp-moderation/screens/block-member-form.php [hunk 6]
```diff
@@ -14,7 +14,9 @@
 				<div class="modal-container">
 					<header class="bb-model-header">
 						<h4><?php esc_html_e( 'Block Member?', 'buddyboss' ); ?></h4>
-						<button title="<?php esc_html_e( 'Close (Esc)', 'buddyboss' ); ?>" type="button" class="mfp-close"></button>
+						<button title="<?php esc_html_e( 'Close (Esc)', 'buddyboss' ); ?>" type="button" class="mfp-close">
+                            <span class="bb-icon bb-icon-close"></span>
+                        </button>
```
Comment (KartikSuthar) [inline src/bp-moderation/screens/block-member-form.php:19]: Fix phpcbf

File: src/bp-moderation/screens/content-report-form.php [hunk 7]
```diff
@@ -13,8 +13,10 @@
 		<div class="modal-wrapper">
 			<div class="modal-container">
 				<header class="bb-model-header">
-					<h4><?php esc_html_e( 'Report Content', 'buddyboss' ); ?></h4>
-					<button title="<?php esc_html_e( 'Close (Esc)', 'buddyboss' ); ?>" type="button" class="mfp-close"></button>
+					<h4><?php esc_html_e( 'Report', 'buddyboss' ); ?> <span class="bp-reported-type"></span></h4>
+					<button title="<?php esc_html_e( 'Close (Esc)', 'buddyboss' ); ?>" type="button" class="mfp-close">
+                        <span class="bb-icon bb-icon-close"></span>
+                    </button>
```
Comment (KartikSuthar) [inline src/bp-moderation/screens/content-report-form.php:19]: Fix phpcbf

File: src/bp-moderation/screens/reported-content-popup.php [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+                    </button>
+				</header>
+
+				<div class="bb-report-type-wrp">
+					<?php printf('%s <span class="bp-reported-type"></span>', esc_html__('You have already reported this', 'buddyboss') ); ?>
+				</div>
+			</div>
+		</div>
+	</div>
+</div>
```
Comment (KartikSuthar) [inline src/bp-moderation/screens/reported-content-popup.php:30]: Fix phpcbf

File: src/bp-templates/bp-nouveau/includes/activity/template-tags.php [hunk 9]
```diff
@@ -836,83 +836,25 @@ function bp_nouveau_get_activity_comment_buttons( $args ) {
...
-			'position'          => 25,
+			'id'                => 'activity_comment_reply',
+			'position'          => 5,
 			'component'         => 'activity',
 			'must_be_logged_in' => true,
 			'parent_element'    => $parent_element,
 			'parent_attr'       => $parent_attr,
 			'button_element'    => $button_element,
-			'link_text'         => __( 'Spam', 'buddyboss' ),
+			'link_text'         => __( 'Reply', 'buddyboss' ),
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/includes/activity/template-tags.php:846]:
```suggestion
			'link_text'         => esc_html__( 'Reply', 'buddyboss' ),
```

File: src/bp-templates/bp-nouveau/includes/activity/template-tags.php [hunk 10]
```diff
@@ -1891,3 +1821,191 @@ function bb_nouveau_get_activity_entry_bubble_buttons( $args ) {
...
+	}
+
+	$buttons['activity_comment_delete'] = array(
+		'id'                => 'activity_comment_delete',
+		'component'         => 'activity',
+		'must_be_logged_in' => true,
+		'parent_element'    => $parent_element,
+		'parent_attr'       => $parent_attr,
+		'button_element'    => $button_element,
+		'link_text'         => __( 'Delete', 'buddyboss' ),
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/includes/activity/template-tags.php:1938]:
```suggestion
		'link_text'         => esc_html__( 'Delete', 'buddyboss' ),
```

File: src/bp-templates/bp-nouveau/includes/groups/template-tags.php [hunk 11]
```diff
@@ -648,456 +687,567 @@ function bp_nouveau_groups_manage_members_buttons( $args = array() ) {
...
-			return $buttons;
-		}
+		// Setup Reject button attributes.
+		$buttons['reject_invite'] = array(
+			'id'                => 'reject_invite',
+			'position'          => 15,
+			'component'         => 'groups',
+			'must_be_logged_in' => true,
+			'parent_element'    => $parent_element,
+			'link_text'         => __( 'Reject', 'buddyboss' ),
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/includes/groups/template-tags.php:793]:
```suggestion
			'link_text'         => esc_html__( 'Reject', 'buddyboss' ),
```

File: src/bp-templates/bp-nouveau/includes/groups/template-tags.php [hunk 12]
```diff
@@ -648,456 +687,567 @@ function bp_nouveau_groups_manage_members_buttons( $args = array() ) {
...
-		if ( ! empty( $args['parent_attr']['class'] ) ) {
-			$parent_class = $args['parent_attr']['class'];
+		$buttons['group_membership_reject'] = array(
+			'id'                => 'group_membership_reject',
+			'position'          => 15,
+			'component'         => 'groups',
+			'must_be_logged_in' => true,
+			'parent_element'    => $parent_element,
+			'button_element'    => $button_element,
+			'link_text'         => __( 'Reject', 'buddyboss' ),
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/includes/groups/template-tags.php:848]:
```suggestion
			'link_text'         => esc_html__( 'Reject', 'buddyboss' ),
```

File: src/bp-templates/bp-nouveau/includes/groups/template-tags.php [hunk 13]
```diff
@@ -648,456 +687,567 @@ function bp_nouveau_groups_manage_members_buttons( $args = array() ) {
...
+			$buttons['group_membership_reject']['button_attr']['href'] = esc_url( bp_get_group_request_reject_link() );
 		}
 
-		// Invite buttons on member's invites screen
-		if ( 'invite' === $type ) {
-			// Don't show button if not logged in or previously banned
-			if ( ! is_user_logged_in() || bp_group_is_user_banned( $group ) || empty( $group->status ) ) {
-				return $buttons;
-			}
+		/*
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/includes/groups/template-tags.php:867]:
```suggestion
		/**
```

File: src/bp-templates/bp-nouveau/includes/groups/template-tags.php [hunk 14]
```diff
@@ -648,456 +687,567 @@ function bp_nouveau_groups_manage_members_buttons( $args = array() ) {
...
+		$buttons = array(
+			'unban_member'  => array(
+				'id'                => 'unban_member',
 				'position'          => 5,
 				'component'         => 'groups',
 				'must_be_logged_in' => true,
 				'parent_element'    => $parent_element,
-				'link_text'         => esc_html__( 'Accept', 'buddyboss' ),
 				'button_element'    => $button_element,
+				'link_text'         => __( 'Remove Ban', 'buddyboss' ),
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/includes/groups/template-tags.php:882]:
```suggestion
				'link_text'         => esc_html__( 'Remove Ban', 'buddyboss' ),
```

File: src/bp-templates/bp-nouveau/includes/groups/template-tags.php [hunk 15]
```diff
@@ -648,456 +687,567 @@ function bp_nouveau_groups_manage_members_buttons( $args = array() ) {
...
+			),
+			'ban_member'    => array(
+				'id'                => 'ban_member',
 				'position'          => 15,
 				'component'         => 'groups',
 				'must_be_logged_in' => true,
 				'parent_element'    => $parent_element,
-				'link_text'         => __( 'Reject', 'buddyboss' ),
+				'button_element'    => $button_element,
+				'link_text'         => __( 'Kick &amp; Ban', 'buddyboss' ),
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/includes/groups/template-tags.php:901]:
```suggestion
				'link_text'         => esc_html__( 'Kick &amp; Ban', 'buddyboss' ),
```

File: src/bp-templates/bp-nouveau/includes/groups/template-tags.php [hunk 16]
```diff
@@ -648,456 +687,567 @@ function bp_nouveau_groups_manage_members_buttons( $args = array() ) {
...
+			),
+			'remove_member' => array(
+				'id'                => 'remove_member',
+				'position'          => 45,
 				'component'         => 'groups',
 				'must_be_logged_in' => true,
 				'parent_element'    => $parent_element,
 				'button_element'    => $button_element,
-				'link_text'         => __( 'Reject', 'buddyboss' ),
+				'link_text'         => __( 'Remove from group', 'buddyboss' ),
```
Comment (KartikSuthar) [inline src/bp-templates/bp-nouveau/includes/groups/template-tags.php:959]:
```suggestion
				'link_text'         => esc_html__( 'Remove from group', 'buddyboss' ),
```

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

**Review by KartikSuthar — APPROVED** [review]
@janakkaneriya007 Merged to dev and pass to QA.

**Comment by mohit-buddyboss** [comment]
@janakkaneriya007  @amincdev 

Tested this on dev branch of the platform and theme. 
Tested with and without theme. 

All changes are implemented successfully and looking fine. PR is ready for release.

connected PR in theme : buddyboss/buddyboss-theme#1343

cc @sourabhmatolia @chetansatasiya

**Pull request merged**